Add examples for QNEthernet

khoih-prog opened the pull request.

ssilverman commented on examples/UdpNtpClient/UdpNtpClient.ino:
```diff
@@ -0,0 +1,174 @@
+  
+  // wait ten seconds before asking for the time again
+  delay(10000);
+  Ethernet.maintain();
```
`Ethernet.maintain()` is no longer needed. I know this is a balancing act, balancing the Arduino API with new features, but I think removing this call is okay.

khoih-prog replied:
It's better to remove this.

ssilverman commented on examples/WebClient/WebClient.ino:
```diff
@@ -0,0 +1,167 @@
+  Serial.println("\nStarting WebClient with NativeEthernet using QNEthernet Library");
+
+  // start the Ethernet connection:
+  Serial.println("Initialize Ethernet with DHCP:");
```
Move comment to both DHCP and static IP sections because "DHCP" shouldn't be printed for both cases.

khoih-prog replied:
Will do like this

```
#if USING_DHCP
  // Start the Ethernet connection, using DHCP
  Serial.print("Initialize Ethernet using DHCP => ");
  Ethernet.begin();
#else  Ethernet.setDNSServerIP(mydnsServer);
  
  // Start the Ethernet connection, using static IP
  Serial.print("Initialize Ethernet using static IP => ");
  Ethernet.begin(myIP, myNetmask, myGW);
#endif
```

ssilverman commented on examples/WebClientRepeating/WebClientRepeating.ino:
```diff
@@ -0,0 +1,145 @@
+  Serial.println("\nStarting WebClientRepeating with NativeEthernet using QNEthernet Library");
+
+  // start the Ethernet connection:
+  Serial.println("Initialize Ethernet with DHCP:");
```
Same "DHCP" comment as above.

khoih-prog replied:
Same as above

ssilverman commented on examples/WebServer/WebServer.ino:
```diff
@@ -0,0 +1,156 @@
+  Serial.println("\nStarting WebServer with NativeEthernet using QNEthernet Library");
+
+  // start the Ethernet connection:
+  Serial.println("Initialize Ethernet with DHCP:");
```
Same "DHCP" comment as above.

khoih-prog replied:
Same as above

ssilverman commented on examples/WebServer/WebServer.ino:
```diff
@@ -0,0 +1,156 @@
+    }
+
+    // give the web browser time to receive the data
+    delay(1);
```
I'd like to see a `flush()` here.

khoih-prog replied:
Do you mean `Serial.flush();` ?

ssilverman commented on examples/WebClient/WebClient.ino:
```diff
@@ -0,0 +1,167 @@
+  Serial.begin(115200);
+  while (!Serial);
+
+  Serial.println("\nStarting WebClient with NativeEthernet using QNEthernet Library");
```
Same "NativeEthernet" question as above.

khoih-prog replied:
Same as above

ssilverman commented on examples/WebClientRepeating/WebClientRepeating.ino:
```diff
@@ -0,0 +1,145 @@
+  Serial.begin(115200);
+  while (!Serial);
+
+  Serial.println("\nStarting WebClientRepeating with NativeEthernet using QNEthernet Library");
```
Same "NativeEthernet" question as above.

khoih-prog replied:
Same as above

ssilverman commented on examples/WebServer/WebServer.ino:
```diff
@@ -0,0 +1,156 @@
+  Serial.begin(115200);
+  while (!Serial);
+
+  Serial.println("\nStarting WebServer with NativeEthernet using QNEthernet Library");
```
Same "NativeEthernet" question as above.

khoih-prog replied:
Same as above

ssilverman reviewed (commented):
Thank you for these examples. I've made a few comments. Would you be okay changing the code style to match the source? I note in one of the comments that I'm trying to be consistent with a single style and I've been using the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html). I also make a note of this in the new "Code style" section of the README. If you're trying to be true to the original source code, I understand, but I might reformat the examples to be "QNEthernet-like" later. I could be convinced to leave them if you'd prefer to keep them as "legacy examples" to show how to adapt old code. That goes for all my comments except for the "NativeEthernet" ones, because I'm not certain those are in the originals.

ssilverman commented:
Would you mind changing the commit comment to pluralize "examples"?

ssilverman commented on examples/UdpNtpClient/UdpNtpClient.ino:
```diff
@@ -0,0 +1,174 @@
+#define USING_DHCP    true
+
+#if !USING_DHCP
+  // Set the static IP address to use if the DHCP fails to assign
```
I don't this comment accurately describes what happens. The variables aren't included if USING_DHCP is true.

Additionally, I'd add a comment requesting the user to assign valid values for their setup.

khoih-prog replied:
Thanks for your comments.

This is !USING_DHCP, not USING_DHCP. To be used only for static IP.
User has to change them to match local network.

ssilverman commented on examples/WebClient/WebClient.ino:
```diff
@@ -0,0 +1,167 @@
+#define USING_DHCP    true
+
+#if !USING_DHCP
+  // Set the static IP address to use if the DHCP fails to assign
```
Same comment as previous example.

khoih-prog replied:
Same as above. !USING_DHCP, not USING_DHCP

ssilverman commented on examples/WebClientRepeating/WebClientRepeating.ino:
```diff
@@ -0,0 +1,145 @@
+#define USING_DHCP    true
+
+#if !USING_DHCP
+// Set the static IP address to use if the DHCP fails to assign
```
Same comment as first example.

khoih-prog replied:
Same as above

ssilverman commented on examples/WebServer/WebServer.ino:
```diff
@@ -0,0 +1,156 @@
+#define USING_DHCP    true
+
+#if !USING_DHCP
+  // Set the static IP address to use if the DHCP fails to assign
```
Same comment as first example.

khoih-prog replied:
Same as above

khoih-prog commented on examples/UdpNtpClient/UdpNtpClient.ino:
```diff
@@ -0,0 +1,174 @@
+  Serial.begin(115200);
+  while (!Serial);
+
+  Serial.println("\nStarting UdpNtpClient with NativeEthernet using QNEthernet Library");
```
That's true. I really don't know how to differentiate between other SPI Ethernet, such as W5500 and this so-called native-Ethernet. We can just remove it now.

khoih-prog commented on examples/UdpNtpClient/UdpNtpClient.ino:
```diff
@@ -0,0 +1,174 @@
+  Ethernet.begin(myIP, myNetmask, myGW);
+#endif
+
+  if (!Ethernet.waitForLocalIP(5000))
```
Please reformat later.

khoih-prog reviewed (commented):
Already modified as requested. Please review

khoih-prog reviewed (commented):
Fix static IP issue for examples now

The call to `Ethernet.setDNSServerIP(mydnsServer)` must be placed after `Ethernet.begin(myIP, myNetmask, myGW)` for the DNS server IP to be valid. Otherwise, `EthernetClient::connect()` will always return **false**

ssilverman commented:
I haven’t forgotten these. I’m hesitating because I want examples that aren’t canonical Arduino style, so I’m toying with my own examples to see what I like and then comparing them with these.

khoih-prog commented:
I understand your goals and priorities. It's OK now as your library's users can now use many other examples from my libraries, ported to support QNEthernet.

The growing list can be found here [**New and Modified Libraries to support Teensy 4.1 QNEthernet**](https://forum.arduino.cc/t/new-and-modified-libraries-to-support-teensy-4-1-qnethernet/902354).

12catarinabarber2930 reviewed (commented):
examples/UdpNtpClient/UdpNtpClient.ino
